core/ledger/kvledger/txmgmt/statedb/statedb.go

```diff
@@ -77,6 +78,12 @@ type VersionedDB interface {
 	Close()
 }
 
+// NamespaceProvider provides a mean to get all the namespaces for a channel. The intended use is for statecouchdb
+// to retroactively build channel metadata when it is missing (e.g., a statecouchdb from an older version)
+type NamespaceProvider interface {
+	GetStateDBNamespaces(channelName string, vdb VersionedDB) ([]string, error)
+}
+
```

Comment:
Thank you @manish-sethi. Agree defining a function may be simpler in some cases. However, will keep the interface for this one, because we need to mock it multiple times in test code, and it is easier to mock an interface.